dev-libs/libaddressinput/libaddressinput-0~git.recipe

```diff
@@ -0,0 +1,85 @@
+SUMMARY="Google’s postal address library that powering Android and Chromium"
+DESCRIPTION="libaddressinput can provide information about \
+what input fields are required for a correct address input form for \
+any country in the world and can validate an address to highlight \
+input errors like missing required fields or invalid values."
+HOMEPAGE="https://github.com/googlei18n/libaddressinput"
+COPYRIGHT="2014 Google Inc."
+LICENSE="Apache v2"
+REVISION="1"
+srcGitRev="a575c63de328d0c72237c55d1a05e191c245122a"
+SOURCE_URI="https://github.com/googlei18n/libaddressinput/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="f92b38a0291372aa97306626f2941485bd1c00bd3cc7e242a1cb9fcd50b62212"
+SOURCE_FILENAME="libaddressinput-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="libaddressinput-$srcGitRev"
+googletestVersion="1.8.0"
+SOURCE_URI_2="https://github.com/google/googletest/archive/release-$googletestVersion.tar.gz"
+CHECKSUM_SHA256_2="58a6f4277ca2bc8565222b3bbd58a177609e9c488e8a72649359ba51450db7d8"
+SOURCE_FILENAME_2="gtest-$googletestVersion.tar.gz"
+SOURCE_URI_3="https://chromium.googlesource.com/chromium/src/tools/grit/+archive/b033674.tar.gz"
+CHECKSUM_SHA256_3="2e9ef06cb237426be60b4031dff82f414bd9a5ded775c2af1fbac51a7fa345a9"
+SOURCE_FILENAME_3="grit.tar.gz"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libaddressinput$secondaryArchSuffix = $portVersion
+	lib:libaddressinput$secondaryArchSuffix = portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libre2
+	"
+
+PROVIDES_devel="
+	libaddressinput${secondaryArchSuffix}_devel
+	devel:libaddressinput$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	libaddressinput$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:rapidjson <= 1.1.0
+	devel:libgtest
+	devel:libre2
+	"
+BUILD_PREREQUIRES="
+	cmd:c++$secondaryArchSuffix
+	cmd:cmp
+	cmd:ninja
+	cmd:gyp
+	"
+
+BUILD()
+{
+	cp -R $sourceDir3/grit-b033674/* externals/grit
+
+	export GYP_DEFINES="
+		gtest_dir='$sourceDir2/googletest-release-$googletestVersion/googletest/include'
+		gtest_src_dir='$sourceDir2/googletest-release-$googletestVersion/googletest'
+		rapidjson_dir='`finddir B_SYSTEM_HEADERS_DIRECTORY`'
+		"
+	export GYP_GENERATORS='ninja'
+	cd cpp
+	gyp --depth .
+	ninja -C out/Default
+}
+
+INSTALL()
+{
+	mkdir -p $libDir
+	cp -a $sourceDir/cpp/out/Default/lib/libaddressinput.so $libDir
+	mkdir -p $includeDir/libaddressinput
+	cp $sourceDir/cpp/include/libaddressinput/* $includeDir/libaddressinput
+	prepareInstalledDevelLib libaddressinput
+	packageEntries devel $developDir
```

Comment:
May I suggest a small variant?
```diff
 INSTALL()
 {
-	mkdir -p $libDir
+	install -d $libDir $includeDir/libaddressinput
-	cp -a $sourceDir/cpp/out/Default/lib/libaddressinput.so $libDir
+	install -t $libDir cpp/out/Default/lib/libaddressinput.so
-	mkdir -p $includeDir/libaddressinput
-	cp $sourceDir/cpp/include/libaddressinput/* $includeDir/libaddressinput
+	install -t $includeDir/libaddressinput cpp/include/libaddressinput/*
+
 	prepareInstalledDevelLib libaddressinput
+	fixPkgconfig
+
-	packageEntries devel $developDir
+	packageEntries devel \
+		$developDir
 }
```
(I haven't tested it but I'm pretty sure it will work.)